LZMA doesn't build with some warnings enabled, and ARM64 hardware SHA512 is not
supported on all of the platforms we build on.

diff --git a/C/Sha512.c b/C/Sha512.c
index 24eba46..9958a04 100755
--- a/C/Sha512.c
+++ b/C/Sha512.c
@@ -20,13 +20,13 @@ This code is based on public domain code from Wei Dai's Crypto++ library. */
   #endif
 #elif defined(MY_CPU_ARM64) && defined(MY_CPU_LE)
   #if defined(__ARM_FEATURE_SHA512)
-    #define Z7_COMPILER_SHA512_SUPPORTED
+    // #define Z7_COMPILER_SHA512_SUPPORTED
   #else
     #if (defined(Z7_CLANG_VERSION) && (Z7_CLANG_VERSION >= 130000) \
            || defined(__GNUC__) && (__GNUC__ >= 9) \
         ) \
       || defined(Z7_MSC_VER_ORIGINAL) && (_MSC_VER >= 1940) // fix it
-      #define Z7_COMPILER_SHA512_SUPPORTED
+      // #define Z7_COMPILER_SHA512_SUPPORTED
     #endif
   #endif
 #endif
diff --git a/C/Sha512Opt.c b/C/Sha512Opt.c
index f4d3703..8c98e1f 100755
--- a/C/Sha512Opt.c
+++ b/C/Sha512Opt.c
@@ -189,7 +189,7 @@ void Z7_FASTCALL Sha512_UpdateBlocks_HW(UInt64 state[8], const Byte *data, size_
            || defined(__GNUC__) && (__GNUC__ >= 9) \
           ) \
       || defined(Z7_MSC_VER_ORIGINAL) && (_MSC_VER >= 1940) // fix it
-      #define USE_HW_SHA
+      // #define USE_HW_SHA
     #endif
   #endif

diff --git a/CPP/7zip/warn_clang.mak b/CPP/7zip/warn_clang.mak
index 01d3d8d..f31b4b1 100755
--- a/CPP/7zip/warn_clang.mak
+++ b/CPP/7zip/warn_clang.mak
@@ -1,3 +1,8 @@
-CFLAGS_WARN = -Weverything -Wfatal-errors
+CFLAGS_WARN = -Weverything -Wfatal-errors \
+  -Wno-unguarded-availability-new \
+  -Wno-implicit-function-declaration \
+  -Wno-poison-system-directories \
+  -Wno-reserved-id-macro \
+  -Wno-unknown-warning-option \
 # CXX_STD_FLAGS = -std=c++11
 # CXX_STD_FLAGS =

